src/providers/ProfileProvider.tsx

```diff
@@ -0,0 +1,31 @@
+'use client';
+import { createContext, type PropsWithChildren } from 'react';
+import { isEVMAddress } from '@/utils/isEVMAddress';
+
+export const ProfileContext = createContext<ProfileProps>({
+  walletAddress: '',
+  isEVMAddress: false,
+  isPublic: false,
+});
+
+interface ProfileProps {
+  walletAddress: string;
+  isPublic?: boolean;
+  isEVMAddress: boolean;
+}
+
+export const ProfileProvider: React.FC<
+  PropsWithChildren<Pick<ProfileProps, 'walletAddress' | 'isPublic'>>
+> = ({ children, walletAddress, isPublic }) => {
+  return (
+    <ProfileContext.Provider
+      value={{
+        walletAddress,
+        isPublic: isPublic ?? false,
+        isEVMAddress: isEVMAddress(walletAddress),
```

Comment:
it should work with Solana address as well, why do we have this boolean?

Reply:
Because I had to extract this information from the useAccount hook (used on some components, mainly effigy related + `useLoyaltyPass` hook (actually surprising that it is used there)